content/ko/docs/concepts/architecture/garbage-collection.md

```diff
@@ -0,0 +1,172 @@
+모든 소유자가 없는 것으로 확인되면 삭제될 수 있다.
+
+클러스터 범위의 종속 오브젝트는 클러스터 범위의 소유자만 지정할 수 있다.
+v1.20 이상에서, 클러스터 범위의 종속 오브젝트가 네임스페이스 종류를 소유자로 지정하면,
+확인할 수 없는 소유자 참조가 있는 것으로 간주되어 가비지 수집이 될 수 없다.
+
+v1.20 이상에서, 가비지 수집기가 잘못된 교차 네임스페이스 `ownerReference`
+또는 네임스페이스 종류를 참조하는 `ownerReference`가 있는 클러스터 범위의 종속 항목을 감지하면,
+`OwnerRefInvalidNamespace`가 원인인 경고 이벤트와 유효하지 않은 종속 항목의 `involvedObject`가 보고된다.
+`kubectl get events -A --field-selector=reason=OwnerRefInvalidNamespace`
+를 실행하여 이러한 종류의 이벤트를 확인할 수 있다.
+{{< /note >}}
+
+## 캐스케이딩(Cascading) 삭제 {#cascading-deletion}
+
+쿠버네티스는 오브젝트를 삭제할 때 더 이상 소유자 참조가 없는지,
+예를 들어 레플리카셋을 삭제할 때, 남겨진 파드가 없는지 확인하고 삭제한다.
+오브젝트를 삭제할 때 쿠버네티스가 오브젝트의 종속 오브젝트들을 자동으로 삭제할 지 여부를 제어할 수 있다.
+이 과정을 `캐스케이딩 삭제`라고 한다.
+캐스케이딩 삭제에는 다음과 같은 두 가지 종류가 있다.
+
+  * 포어그라운드 캐스케이딩 삭제(Foreground cascading deletion)
+  * 백그라운드 캐스케이딩 삭제(Background cascading deletion)
+
+또한 쿠버네티스의 {{<glossary_tooltip text="finalizers" term_id="finalizer">}}를 사용하여 가비지 수집이 소유자 참조가 있는 자원을 언제 어떻게 삭제할 것인지 제어할 수 있다.
+
+### 포어그라운드 캐스케이딩 삭제 {#foreground-deletion}
+
+포어그라운드 캐스케이딩 삭제에서는 삭제하려는 소유자 오브젝트가 먼저
+*삭제 중* 상태가 된다. 이 상태에서는 소유자 오브젝트에게 다음과 같은 일이
+일어난다:
+
+  * 쿠버네티스 API 서버가 오브젝트의 `metadata.deletionTimestamp` 필드를
+    오브젝트가 삭제 표시된 시간으로 설정한다.
+  * 쿠버네티스 API 서버가 `metadata.finalizers` 필드를 `foregroundDeletion`로
+    설정한다. 
+  * 오브젝트는 삭제 과정이 완료되기 전까지 쿠버네티스 API를 통해 조회할 수 있다.
+
+소유자 오브젝트가 삭제 중 상태가 된 이후, 컨트롤러는 종속 오브젝트들을 삭제한다.
+모든 종속 오브젝트들이 삭제되고나면, 컨트롤러가 소유자 오브젝트를 삭제한다.
+이 시점에서 오브젝트는 더 이상 쿠버네티스 API를 통해 조회할 수 없다.
+
+포어그라운드 캐스케이딩 삭제 중에 소유자 오브젝트의 삭제를 막는
+종속 오브젝트는`ownerReference.blockOwnerDeletion=true`필드를 가진 오브젝트다.
+더 자세한 내용은 [Use foreground cascading deletion](/docs/tasks/administer-cluster/use-cascading-deletion/#use-foreground-cascading-deletion)를
+참고한다.
+
+### 백그라운드 캐스케이딩 삭제 {#background-deletion}
+
+백그라운드 캐스케이딩 삭제에서는 쿠버네티스 API 서버가 소유자 오브젝트를 즉시 삭제하고
+백그라운드에서 컨트롤러가 종속 오브젝트들을 삭제한다.
+쿠버네티스는 수동으로 포어그라운드 삭제를 사용하거나 종속 오브젝트를 분리하지 않는다면, 기본적으로 백그라운드 캐스케이딩 삭제를 사용한다.
+
+더 자세한 내용은 [Use background cascading deletion](/docs/tasks/administer-cluster/use-cascading-deletion/#use-background-cascading-deletion)를
+참고한다.
+
+### 분리된 종속 (Orphaned dependents)
+
+쿠버네티스가 소유자 오브젝트를 삭제할 때, 남은 종속 오브젝트는 *분리된* 오브젝트라고 부른다.
+기본적으로 쿠버네티스는 종속 오브젝트를 삭제한다.
+이 행동을 오버라이드하는 방법을 보려면,
+다음 [Delete owner objects and orphan dependents](/docs/tasks/administer-cluster/use-cascading-deletion/#set-orphan-deletion-policy)를 참고한다.
+
+## 사용되지 않는 컨테이너와 이미지 가비지 수집 {#containers-images}
+
+{{<glossary_tooltip text="kubelet" term_id="kubelet">}}은
+사용되지 않는 이미지에 대한 가비지 수집을 5분마다, 컨테이너에 대한 가비지 수집을 1분마다
+수행한다. 외부 가비지 수집 도구는 Kubelet 의 행동을 중단시키고
+존재해야만 하는 컨테이너를 삭제할 수 있으므로 사용을 피해야 한다.
+
+사용되지 않는 컨테이너와 이미지에 대한 가비지 수집 옵션을 구성하려면, 
+[configuration file](/docs/tasks/administer-cluster/kubelet-config-file/) 사용하여 Kubelet 을 수정하거나
+[`KubeletConfiguration`](/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubeletConfiguration) 리소스 타입의
+가비지 수집과 관련된 파라미터를 수정한다.
+
+### 컨테이너 이미지 라이프사이클
+
+쿠버네티스는 Kubelet의 일부인 *이미지 관리자*가 cadvisor와 협동하여
```

Comment:
Q: preserving "{{< glossary_tooltip text="cadvisor" term_id="cadvisor" >}}" from English is better?

Reply:
It'd be better :)